.github/workflows/push-to-master.yml

```diff
@@ -1,8 +1,5 @@
 name: push-to-master
-on:
-  push:
-    branches:
-      - master
+on: pull_request
```

Comment:
wouldn't it be better to also run the doc-checker job on `pull-request` trigger to see whether the PR broke some docs?

Reply:
Тot sure if I understood you correctly, doc-checker job [was ran](https://github.com/modin-project/modin/pull/3237/checks?check_run_id=3058042859) (it declared in the `.github/workflows/ci.yml` that is triggered on `pull_request`).

Reply:
Oh, ok, then why this change is needed?

Reply:
This change is needed because bug was detected by the test declared in the `.github/workflows/push-to-master.yml` (that is triggered on the push to the master action) and in order to make sure that our changes fix the issue, we are running these tests during PR review (after PR approve we will rollback this CI job change and merge only fix change).

Reply:
We both get confused :), when I said "doc-checker" I meant "test-doc" - that particular job which failed on master, but wasn't visible at the PR' jobs. What I'm proposing is to also copy the "test-doc" job for `pull-request` trigger, so we could detect the problematic docs at the PR stage.

Reply:
Oh, looks like it is) But this job was ran on this PR ([passed PR `test-doc` CI job](https://github.com/modin-project/modin/pull/3237/checks?check_run_id=3058042127), [failed master `test-doc` CI job](https://github.com/modin-project/modin/runs/3056230303)). All `.github/workflows/push-to-master.yml` jobs execute at once (changed `on` parameter affects all jobs in the `.github/workflows/push-to-master.yml` config).

Reply:
I mean for the future bug detection. I proposed to move "test-doc" to the "pull-request" trigger permanently

Reply:
I see now) Previously `test-doc` job was moved exactly on "pull-request" trigger to reduce the load on pandas docs server (https://github.com/modin-project/modin/issues/3003), now sure we should revert it.

Reply:
discussed offline and decided to left things as it is for now